src/collective/ifttt/browser/views.py

```diff
@@ -24,3 +32,38 @@ def check_iftttconfig(self):
         if not secret_key:
             return False
         return True
+
+
+@provider(IContextSourceBinder)
+def availableTriggers(context):
+
+    # sometimes context is a collection so,
+    # we need to traverse to it's parent folder
+    context = context
+    allowed_portal_type = ['Folder', 'Plone Site']
+    while context.portal_type not in allowed_portal_type:
+        context = aq_parent(context)
```

Comment:
@Shriyanshagro Wouldn't it be better to completely hide IFTTT menu when the context is Collection?

Reply:
Oh, so it points to default content... that must be fixed in the menu:

"context_state" has function to get "canonical_object" that means the folder instead of its default page. I believe we can get that and then use "context_state" for it:

https://github.com/collective/collective.ifttt/blob/master/src/collective/ifttt/browser/menu.py#L47

```
context_state = getMultiAdapter((context, request), name='plone_context_state')
context_state = getMultiAdapter((context_state.canonical_object(), request), name='plone_context_state')
```